### ISSUE TITLE
[tune] Setting memory limit for trial

### ISSUE THREAD

**Issue by moritzmeister** [ray-project/ray#10640]
I would like to run an experiment with Tune in comparison to another Distributed Hyperparameter Tuning tool, and in order to guarantee comparibility, I would like to set a maximum amount of memory to be used for a trial.

Is this possible?
I couldn't find an example about how to set the memory in the `resources_per_trial` argument.

Otherwise, how does Tune allocate memory among trials? Does it just split all available memory among the N concurrent trials?

Is there a section in the documentation about this that I missed?

Thanks for the help

**Comment by richardliaw** [comment]
Hi Moritz, I think the closest you can do is set `object_store_memory` or `memory` in `resources_per_trial`. 

As far as I remember, I don't think we actually "enforce" memory limits (as in, we expect trials to be good citizens and abide by the memory limits).

This might be useful: 
https://github.com/ray-project/ray/blob/39c598bab0929f007a49daa7262a7c9ee6492221/python/ray/tests/test_memory_scheduling.py

**Comment by moritzmeister** [comment]
Thanks! That will do :) 
I think the `memory` argument for `resources_per_trial` is not very well documented in the *tune* docs though.

**Comment by richardliaw** [comment]
Agree. Let me know what comes out of it! I'll label this as a documentation TODO for now.
